src/app/common/constants.ts

```diff
@@ -38,22 +38,21 @@ export enum StatusCategoryType {
   Warn = 'Warn',
 }
 
-export enum PlanStatusType {
-  Ready = 'Ready',
-  Executing = 'Executing',
-  Succeeded = 'Succeeded',
-  Failed = 'Failed',
-  Canceled = 'Canceled',
-}
-
-export enum PlanStatusDisplayType {
-  Ready = 'Ready',
-  Executing = 'Running',
-  Succeeded = 'Succeeded',
-  Canceled = 'Canceled',
-  Failed = 'Failed',
-  Pending = 'Pending',
-}
+export type ConditionType = 'Ready' | 'Executing' | 'Succeeded' | 'Failed' | 'Canceled';
+
+export type ColdPlanState = Exclude<PlanState, 'Copying' | 'AbortedCopying' | 'StartingCutover'>;
+
+export type PlanState =
+  | 'NotStarted' // Just created, no action taken yet
+  | 'Starting' // User clicked start button, Migration CR exists but no status data exists yet
+  | 'Copying' // Warm-specific. "Running incremental copies" in the UI, the first stage of a warm migration
+  | 'AbortedCopying' // Warm-specific. Plan was canceled during copying
```

Comment:
I think we need to rename `AbortedCopying` to `FailedCopying` and add a new state `CanceledCopying`

Reply:
Making all these states explicit seems like a great improvement, thanks for all the help!